armi/physics/neutronics/fissionProductModel/lumpedFissionProduct.py

```diff
@@ -580,28 +398,50 @@ def _readOneLFP(self, linesOfOneLFP):
             "Loaded {0} {1} nuclides for a total yield of {2}"
             "".format(len(lfp.yld), lfp.name, totalYield)
         )
-
         return lfp
 
 
 def lumpedFissionProductFactory(cs):
     """Build lumped fission products."""
+    if cs["fpModel"] == "explicitFissionProducts":
+        runLog.info("Fission products will be modeled explicitly.")
+        return None
+
     if cs["fpModel"] == "MO99":
-        runLog.warning(
+        runLog.info(
             "Activating MO99-fission product model. All FPs are treated a MO99!"
         )
         return _buildMo99LumpedFissionProduct()
 
     lfpPath = cs[CONF_LFP_COMPOSITION_FILE_PATH]
-    if not lfpPath:
-        return None
+    if not lfpPath or not os.path.exists(lfpPath):
+        raise ValueError(
+            f"The fission product reference file does "
+            f"not exist or is not a valid path. Path provided: {lfpPath}"
+        )
+
     runLog.extra(f"Loading global lumped fission products (LFPs) from {lfpPath}")
     with open(lfpPath) as lfpStream:
         lfpFile = FissionProductDefinitionFile(lfpStream)
         lfps = lfpFile.createLFPsFromFile()
     return lfps
 
 
+def getAllNuclideBasesByLibrary(cs):
+    """Return a list of nuclide bases that are available for a given `fpModelLibrary`."""
+    nbs = []
+    if cs["fpModel"] == "explicitFissionProducts":
+        if cs["fpModelLibrary"] == "MC2-3":
+            nbs = nuclideBases.byMcc3Id.values()
+        else:
+            raise ValueError(
+                f"An option to handle the `fpModelLibrary` "
+                f"set to `{cs['fpModelLibrary']}` has not been "
+                f"implemented."
+            )
+    return nbs
+
+
```

Comment:
This function doesn't appear to belong in this module.

Reply:
Yeah I can move.

Reply:
Moved to `isotopicOptions.py`